dev/ppc64le/src/intt_ppc.S

```diff
@@ -0,0 +1,664 @@
+	BREDUCE_4X 4, 9, 13, 17
+	Write_B4C 32+4, 32+9, 32+13, 32+17
+	Set_mont_consts
+	lvx	V_ZETA, 0, 14
+	addi	14, 14, 16
+	MREDUCE_4X V_ZETA, V_ZETA, V_ZETA, V_ZETA, 13, 18, 23, 28
+	Write_M4C 32+13, 32+18, 32+23, 32+28
+	li	5, 64
+
+	Load_4Coeffs 5, 16, 16
+	BREDUCE_4X 4, 9, 13, 17
+	Write_B4C 32+4, 32+9, 32+13, 32+17
+	Set_mont_consts
+	lxv	32+10, -16(14)
+	MREDUCE_4X V_ZETA, V_ZETA, V_ZETA, V_ZETA, 13, 18, 23, 28
+	Write_M4C 32+13, 32+18, 32+23, 32+28
+	li	5, 256
+
+	Load_4Coeffs 5, 16, 16
+	BREDUCE_4X 4, 9, 13, 17
+	Write_B4C 32+4, 32+9, 32+13, 32+17
+	Set_mont_consts
+	lvx	V_ZETA, 0, 14
+	addi	14, 14, 16
+	MREDUCE_4X V_ZETA, V_ZETA, V_ZETA, V_ZETA, 13, 18, 23, 28
+	Write_M4C 32+13, 32+18, 32+23, 32+28
+	li	5, 320
+
+	Load_4Coeffs 5, 16, 16
+	BREDUCE_4X 4, 9, 13, 17
+	Write_B4C 32+4, 32+9, 32+13, 32+17
+	Set_mont_consts
+	lxv	32+10, -16(14)
+	MREDUCE_4X V_ZETA, V_ZETA, V_ZETA, V_ZETA, 13, 18, 23, 28
+	Write_M4C 32+13, 32+18, 32+23, 32+28
+
+.align 4
+	# 7. len = 128, start = 0
+	#
+	#addi	14, 14, 992
+	li	5, 0            # start
+	li	7, 256          # len * 2
+
+	Load_4Coeffs 5, 16, 16
+	BREDUCE_4X 4, 9, 13, 17
+	Write_B4C 32+4, 32+9, 32+13, 32+17
+	Set_mont_consts
+	lvx	V_ZETA, 0, 14
+	xxlor	9, 32+10, 32+10
+	MREDUCE_4X V_ZETA, V_ZETA, V_ZETA, V_ZETA, 13, 18, 23, 28
+	Write_M4C 32+13, 32+18, 32+23, 32+28
+	li	5, 64
+
+	Load_4Coeffs 5, 16, 16
+	BREDUCE_4X 4, 9, 13, 17
+	Write_B4C 32+4, 32+9, 32+13, 32+17
+	Set_mont_consts
+	xxlor	32+10, 9, 9
+	MREDUCE_4X V_ZETA, V_ZETA, V_ZETA, V_ZETA, 13, 18, 23, 28
+	Write_M4C 32+13, 32+18, 32+23, 32+28
+	li	5, 128
+
+	Load_4Coeffs 5, 16, 16
+	BREDUCE_4X 4, 9, 13, 17
+	Write_B4C 32+4, 32+9, 32+13, 32+17
+	Set_mont_consts
+	xxlor	32+10, 9, 9
+	MREDUCE_4X V_ZETA, V_ZETA, V_ZETA, V_ZETA, 13, 18, 23, 28
+	Write_M4C 32+13, 32+18, 32+23, 32+28
+	li	5, 192
+
+	Load_4Coeffs 5, 16, 16
+	BREDUCE_4X 4, 9, 13, 17
+	Write_B4C 32+4, 32+9, 32+13, 32+17
+	Set_mont_consts
+	xxlor	32+10, 9, 9
+	MREDUCE_4X V_ZETA, V_ZETA, V_ZETA, V_ZETA, 13, 18, 23, 28
+	Write_M4C 32+13, 32+18, 32+23, 32+28
+
+.align 4
+	#
+	# Montgomery reduce loops with constant 1441
+	#
+	addi	14, 4, C1441_OFFSET
+	lvx	V1441, 0, 14
+
+	Reload_4coeffs
+	MREDUCE_4X V1441, V1441, V1441, V1441, 6, 7, 8, 9
+	Reload_4coeffs
+	MREDUCE_4X V1441, V1441, V1441, V1441, 13, 18, 23, 28
+	MWrite_8X 32+6, 32+7, 32+8, 32+9, 32+13, 32+18, 32+23, 32+28
+
+	Reload_4coeffs
+	MREDUCE_4X V1441, V1441, V1441, V1441, 6, 7, 8, 9
+	Reload_4coeffs
+	MREDUCE_4X V1441, V1441, V1441, V1441, 13, 18, 23, 28
+	MWrite_8X 32+6, 32+7, 32+8, 32+9, 32+13, 32+18, 32+23, 32+28
+
+	Reload_4coeffs
+	MREDUCE_4X V1441, V1441, V1441, V1441, 6, 7, 8, 9
+	Reload_4coeffs
+	MREDUCE_4X V1441, V1441, V1441, V1441, 13, 18, 23, 28
+	MWrite_8X 32+6, 32+7, 32+8, 32+9, 32+13, 32+18, 32+23, 32+28
+
+	Reload_4coeffs
+	MREDUCE_4X V1441, V1441, V1441, V1441, 6, 7, 8, 9
+	Reload_4coeffs
+	MREDUCE_4X V1441, V1441, V1441, V1441, 13, 18, 23, 28
+	MWrite_8X 32+6, 32+7, 32+8, 32+9, 32+13, 32+18, 32+23, 32+28
```

Comment:
@dannytsen This will need to be moved to the beginning of the invNTT: We don't reduce the output of the base multiplication, which means that input coefficient to the invNTT is essentially unconstrained in size. Frontloading the Montgomery scaling acts as a reduction.